src/shared/telemetry/telemetryEvent.ts

```diff
@@ -7,21 +7,20 @@ import { MetadataEntry, MetricDatum } from './clienttelemetry'
 import { Datum } from './telemetryTypes'
 
 const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:]', 'g')
-const REMOVE_UNDERSCORES_REGEX = new RegExp('_', 'g')
 
 export interface TelemetryEvent {
-    namespace: string
     createTime: Date
-    data?: Datum[]
+    data: Datum[]
 }
 
 export function toMetricData(array: TelemetryEvent[]): MetricDatum[] {
     return ([] as MetricDatum[]).concat(
-        ...array.map(metricEvent => {
-            const namespace = metricEvent.namespace.replace(REMOVE_UNDERSCORES_REGEX, '')
-
-            if (metricEvent.data !== undefined) {
-                const mappedEventData = metricEvent.data.map(datum => {
+        ...array
+            .filter(item => {
```

Comment:
I'm not sure what we're filtering out, should we still send that data in some capacity? It might inform us that we haven't crafted some events correctly

Reply:
We are filtering out metrics that only have a namespace, which made no sense. Eventually it should be set to not be nullable.